### DISCUSSION TITLE
Any API to automate publish .spoke files

### DISCUSSION THREAD

**Discussion by davegoopot** [mozilla/hubs#3116] [Spoke Questions] [answered]
Is there an API/http end point I can use to publish legacy .spoke files directly?

I've a bunch of .spoke files that I'd like to publish. I'm currently doing this manually from the web user interface as: import legacy .spoke; publish to hubs

**Comment by robinkwilson** [comment, marked as answer]
Currently I don't think there's an API to do this automatically. You can create a issue as a feature request in the Spoke Repo: https://github.com/mozilla/Spoke/issues

**Reply by robertlong** [reply]
Yeah, there's currently no API to publish projects right now. You can automate the creation of projects from a spoke file though.

This endpoint will create a project using a spoke file hosted at a given url:
`https://hubs.mozilla.com/spoke/projects/new?template=<spokefileurl>`

Example:
`https://hubs.mozilla.com/spoke/projects/new?template=https://hubs-proxy.com/https://raw.githubusercontent.com/mozilla/Spoke/master/src/assets/templates/moz-atrium.spoke`

**Reply by davegoopot** [reply]
Super. That's what I need. Thanks.
